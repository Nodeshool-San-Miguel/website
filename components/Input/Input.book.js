import React from "react"
import Input from "./index"
import page from "./docs.mdx"

export default {
  title: "Input",
  parameters: {
    docs: {
      page
    },
    backgrounds: {
      values: [{ name: "dark", value: "#000" }]
    }
  },
  component: Input,
  argTypes: { onChange: { action: "onChange" } }
}

export const Default = (args) => <Input {...args} />

export const InputDark = (args) => <Input.Dark {...args} />
InputDark.storyName = "Input.Dark"

export const InputLight = (args) => <Input.Light {...args} />
InputLight.storyName = "Input.Light"
InputLight.parameters = {
  backgrounds: { default: "dark" }
}

export const WithLabel = (args) => <Input label="Custom label" {...args} />

export const WithPlaceholder = (args) => (
  <Input placeholder="Escribe Átomico" {...args} />
)

export const LightWithLabelAndPlaceholder = (args) => (
  <Input.Light placeholder="123-321" label="Sup!" {...args} />
)
LightWithLabelAndPlaceholder.parameters = {
  backgrounds: { default: "dark" }
}
