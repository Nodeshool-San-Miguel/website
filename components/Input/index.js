import React from "react"
import types from "prop-types"
/**
 * @typedef Input
 * @property { (text: String) => null } onChange - onChange handler callbacks with `.value`
 * @property { String } label - The input label, defaults to "Digita algo"
 * @property { String } className - Extra className for <input/>
 * @property { String } labelClassName - Extra className for label
 * @property { String } containerClassName - Extra className for input container
 * @property { String? } value - Controlled component value
 */
const propTypes = {
  label: types.string,
  placeholder: types.string,
  className: types.string,
  labelClassName: types.string,
  containerClassName: types.string,
  onChange: types.func,
  value: types.string
}
const Template = ({
  label = "Digita algo",
  placeholder = "Digita algo...",
  className = "",
  labelClassName = "",
  containerClassName = "",
  onChange = () => null,
  value = undefined,
  theme
}) => {
  return (
    <div className={`w-full ${containerClassName}`}>
      <div className={`text-xs font-poppins ${theme.label} ${labelClassName}`}>
        {label}
      </div>
      <input
        onChange={(e) => onChange(e.target.value)}
        placeholder={placeholder}
        type="text"
        value={value}
        className={`${theme.input} bg-transparent font-poppins border-b-2 px-4 py-2 w-full ${className}`}
      />
    </div>
  )
}
Template.propTypes = {
  ...propTypes,
  theme: types.shape({
    label: types.string,
    input: types.string
  }).isRequired
}

/**
 * Light flavor Input
 * @param { Input } props
 */
function Light(props) {
  return (
    <Template
      {...props}
      theme={{
        label: "text-black",
        input: "border-dark-15"
      }}
    />
  )
}
Light.propTypes = propTypes

/**
 * Dark flavor Input
 * @param { Input } props
 */
function Dark(props) {
  return (
    <Template
      {...props}
      theme={{
        label: "text-white",
        input: "border-light-15 text-white"
      }}
    />
  )
}
Dark.propTypes = propTypes

/**
 * Light flavor Input
 * @param { Input } props
 */
function Input(props) {
  return <Light {...props} />
}

Input.Light = Light
Input.Dark = Dark
Input.propTypes = propTypes
export default Input
