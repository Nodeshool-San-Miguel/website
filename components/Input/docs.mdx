import Input from "./index"

# Input Component

## Cómo usar

El componente input por default retorna el Sabor, `Dark`

<div className="bg-white p-4">
  <Input value="Alo" />
</div>

También esta el sablr `Light`

<div className="bg-black p-4">
  <Input.Light value="Alo" />
</div>

```js
<Input.Light value="Alo" />
```

Puedes agregar onChange controller, también puedes controlar el value del valor con la prop `value`,
puedes agregar clases personalizadas al contenedor, label y input del componente.

## Importando el componente

```js
import Input from "../components/Input"
// Desde pages/index
```

```js
import Input from "@components/Input"
// Con path aliases
```

## Sabores (Flavors/Types)

```js
<Input.Dark/>
<Input.Light/>
```
