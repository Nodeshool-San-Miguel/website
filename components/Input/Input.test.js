import React from "react"
import Input from "./index"
const rr = require("react-test-renderer")

const render = rr
  .create(<Input value="one" placeholder="two" label="three" />)
  .toJSON()

it("Matches snapshot", () => {
  expect(render).toMatchSnapshot()
})
