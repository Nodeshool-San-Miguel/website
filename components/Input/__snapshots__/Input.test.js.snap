// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="w-full "
>
  <div
    className="text-xs font-poppins text-black "
  >
    three
  </div>
  <input
    className="border-dark-15 bg-transparent font-poppins border-b-2 px-4 py-2 w-full "
    onChange={[Function]}
    placeholder="two"
    type="text"
    value="one"
  />
</div>
`;
